rest/route/project.go

```diff
@@ -748,3 +750,76 @@ func (h *getProjectVersionsHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return resp
 }
+
+type testProjectAliasHandler struct {
+	sc data.Connector
+
+	version             string
+	alias               string
+	includeDependencies bool
+	project             *dbModel.Project
+}
+
+func makeTestProjectAliasHandler(sc data.Connector) gimlet.RouteHandler {
+	return &testProjectAliasHandler{
+		sc: sc,
+	}
+}
+
+func (h *testProjectAliasHandler) Factory() gimlet.RouteHandler {
+	return &testProjectAliasHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *testProjectAliasHandler) Parse(ctx context.Context, r *http.Request) error {
+	params := r.URL.Query()
+
+	h.version = params.Get("version")
+	if h.version == "" {
+		return errors.New("'version' parameter must be specified")
+	}
+	h.alias = params.Get("alias")
+	if h.alias == "" {
+		return errors.New("'alias' parameter must be specified")
+	}
+	h.includeDependencies = (params.Get("include_deps") != "")
+
+	v, err := dbModel.VersionFindOneId(h.version)
+	if err != nil {
+		return errors.Errorf("version %s not found", h.version)
+	}
+	pp, err := dbModel.ParserProjectFindOneById(h.version)
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "error getting parser project",
+		}))
+		return errors.New("unexpected error when retrieving version config")
```

Comment:
Why are we logging but not returning errors?

Reply:
To avoid returning db errors to the client since consumers won't know what it means

Reply:
Hmm we don't do this in other routes though. Isn't it still helpful for them to get the error so they can ask us about it? Or at least this doesn't feel consistent, since a couple lines down we _do_ return the full error and that also has the potential to be a db error, so if you do want to hide db errors then that one should be hidden too.